server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -595,15 +595,15 @@ public RestoreInfo getRestoreInfo() {
     }
 
     public static class RestoreInProgressUpdater extends RoutingChangesObserver.AbstractRoutingChangesObserver {
-        private final Map<String, Updates> shardChanges = new HashMap<>();
+        private final Map<String, Map<ShardId, ShardRestoreStatus>> shardChanges = new HashMap<>();
```

Comment:
In general, I don't like nested maps. Here it's ok, but I'd like to have a comment on what is inside. Especially for such generic type as String.